### ISSUE TITLE
[Submission] 5 params — JagNL (hand-coded)

### ISSUE THREAD

**Issue by JagNL** [anadim/AdderBoard#44]
### Author

JagNL

### Unique Parameter Count

5

### Accuracy

100.00% on 10,000 random pairs + 10 edge cases (seed=2025)

### Method

Hand-coded weights

### Architecture

2L decoder, d=5→16, 5h+1h, ALiBi slope=log(10)

### Key Tricks

- 5 unique scalar params: BASE=10, K_WEIGHT=960, K_BIAS=-1000, V_W1=0.1, DIGIT_OFFSET=0.5
- All weights derived from these 5 (embedding, projections, FFN all computed)
- ALiBi with slope=log(10) for base-10 positional weighting
- Sparse embedding (13 nonzero entries from BASE×digit)
- softmax1 (+1 denominator) for selective attention gating
- Gated ReLU FFN with positional place-value scaling
- V-shape ReLU (|x|) for digit discrimination in L1 FFN
- Parabolic logit decode: -(h/scale)²×100, correct digit ≈ 0 → max logit
- float64 throughout for numerical stability

### Link to Code

https://gist.github.com/JagNL/72f25f6947751cd327491832c225a95a

### Verification Output

```shell
Model: TinyAdder-5
Author: JagNL
Parameters (unique): 5
Architecture: 2L decoder, d=5→16, 5h+1h, ALiBi slope=log(10)
Tricks: 5 unique scalar params, all weights derived from 5 params, ALiBi slope=log(10), sparse embedding, softmax1, gated ReLU FFN, V-shape ReLU, parabolic logit decode, float64

  Progress: 1000/10010 (1000/1000 correct) [8.6s]
  Progress: 2000/10010 (2000/2000 correct) [17.2s]
  Progress: 3000/10010 (3000/3000 correct) [25.7s]
  Progress: 4000/10010 (4000/4000 correct) [34.3s]
  Progress: 5000/10010 (5000/5000 correct) [42.7s]
  Progress: 6000/10010 (6000/6000 correct) [51.4s]
  Progress: 7000/10010 (7000/7000 correct) [59.9s]
  Progress: 8000/10010 (8000/8000 correct) [67.7s]
  Progress: 9000/10010 (9000/9000 correct) [76.0s]
  Progress: 10000/10010 (10000/10000 correct) [84.6s]

Results: 10010/10010 correct (100.00%)
Time: 84.6s (118 additions/sec)
Status: QUALIFIED (threshold: 99%)

### Additional Notes

Constructive proof that 10-digit addition can be represented with 5 unique scalar values in a genuine autoregressive transformer. All embedding entries, projection weights, and FFN coefficients are deterministically derived from these 5 parameters. Generic decode_greedy loop — no addition-specific logic in inference code.
```
